### DISCUSSION TITLE
Why is `building:levels` required to add `roof:shape`? [change implemented]

### DISCUSSION THREAD

**Discussion by legofahrrad** [streetcomplete/StreetComplete#4261] [Q&A] [answered]
To add the roof shape of a building, `building:levels` or `roof_levels` must be set. I wondered why.

https://github.com/streetcomplete/StreetComplete/blob/cf9d62ec470817be163c93b78d84111420d625ad/app/src/main/java/de/westnordost/streetcomplete/quests/roof_shape/AddRoofShape.kt#L21-L24

I came across this issue when I was outside and wanted to add roof shapes but the building level quest was showing up although I gave the roof shape quest a higher priority in the settings.

Maybe someone can explain ;)
Thank you!

**Comment by Helium314** [comment]
See #2556, where this was added.
Maybe it could be different now that the quest is disabled by default.

**Reply by matkoniecz** [reply]
The question is whether it should be enabled for all buildings or for all buildings which are eligible for `building:levels` quest.

For example, should it be asked for `building=garage`?

**Reply by Helium314** [reply]
Is the roof shape any more or less valuable depending on the building?

**Reply by matkoniecz** [reply]
I expect it to be less useful to ask for example for `building=garage` where flat can be assumed more reliably than on `building=house`

Asking for `building=church` is more likely to result in confused mapper and answer "many different" which is not so useful.

**Reply by legofahrrad** [reply]
I would propose to ask the roof shape on the same buildings as the building levels quest.

**Reply by legofahrrad** [reply]
https://github.com/streetcomplete/StreetComplete/blob/cf9d62ec470817be163c93b78d84111420d625ad/app/src/main/java/de/westnordost/streetcomplete/quests/building_levels/AddBuildingLevels.kt#L36-L41

Maybe skip "parking" as these have mostly flat roofs for rooftop parking.

**Comment by matkoniecz** [comment, marked as answer]
Now roof shape is shown where building levels is not yet tagged but eligible https://github.com/streetcomplete/StreetComplete/commit/448d05a0ea0fd00f42f5d9efb7f0f779016f09ca

Not yet released. Will affect only people who changed quest order putting roof shape before building levels (or disabled building levels quest) AND enabled roof shape quest AND are not in country where roofs are usually flat.

Change not yet released, just committed.

**Reply by legofahrrad** [reply]
Great, thank you 🙂
